src/reconcile/rec_write.c

```diff
@@ -2043,6 +2044,222 @@ __rec_compression_adjust(WT_SESSION_IMPL *session, uint32_t max, size_t compress
     WT_WRITE_ONCE(*adjustp, new);
 }
 
+/*
+ * __rec_build_delta_init --
+ *     Build delta init.
+ */
+static int
+__rec_build_delta_init(WT_SESSION_IMPL *session, WT_RECONCILE *r)
+{
+    WT_RET(__wt_buf_init(session, &r->delta, r->disk_img_buf_size));
+    memset(r->delta.mem, 0, WT_PAGE_DELTA_HEADER_SIZE);
+    r->delta.size = WT_PAGE_DELTA_HEADER_BYTE_SIZE(S2BT(session));
+
+    return (0);
+}
+
+/*
+ * __rec_delta_pack_key --
+ *     Pack the delta key
+ */
+static WT_INLINE int
+__rec_delta_pack_key(WT_SESSION_IMPL *session, WT_BTREE *btree, WT_RECONCILE *r, WT_INSERT *ins,
+  WT_ROW *rip, WT_ITEM *key)
+{
+    WT_DECL_RET;
+    uint8_t *p;
+
+    switch (r->page->type) {
+    case WT_PAGE_COL_FIX:
+    case WT_PAGE_COL_VAR:
+        p = key->mem;
+        WT_RET(__wt_vpack_uint(&p, 0, WT_INSERT_RECNO(ins)));
+        key->size = WT_PTRDIFF(p, key->data);
+        break;
+    case WT_PAGE_ROW_LEAF:
+        if (ins == NULL) {
+            WT_WITH_BTREE(
+              session, btree, ret = __wt_row_leaf_key(session, r->page, rip, key, false));
+            WT_RET(ret);
+        } else {
+            key->data = WT_INSERT_KEY(ins);
+            key->size = WT_INSERT_KEY_SIZE(ins);
+        }
+        break;
+    default:
+        WT_RET(__wt_illegal_value(session, r->page->type));
+    }
+
+    return (ret);
+}
+
+/*
+ * __rec_pack_delta_leaf --
+ *     Pack a delta for a leaf page
+ */
+static int
+__rec_pack_delta_leaf(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_SAVE_UPD *supd)
+{
+    WT_DECL_RET;
+    WT_ITEM *key;
+    size_t max_packed_size;
+    uint8_t flags;
+    uint8_t *p, *head;
+
+    flags = 0;
+
+    /* Ensure enough room for a column-store key without checking. */
+    WT_RET(__wt_scr_alloc(session, WT_INTPACK64_MAXSIZE, &key));
+
+    WT_ERR(__rec_delta_pack_key(session, S2BT(session), r, supd->ins, supd->rip, key));
+
+    /*
+     * The max length of a delta:
+     * 1 header byte
+     * 4 timestamps (4 * 9)
+     * key size (5)
+     * key
+     * value size (5)
+     * value
+     */
```

Comment:
This comment defines the shape of the delta.  I would suggest having something a little more complete to describe this, maybe in btmem.h where WT_DELTA_HEADER is defined.  Something like:

A delta is a durable record of changes to a page since the previous delta or full-page image was recorded.  It is composed of a WT_DELTA_HEADER, followed by a variable number of delta entries.  The number of entries does not explicitly appear, it is inferred by the size of the overall delta.

Each delta entry has a one header 'flags' byte (flags from WT_DELTA_CELL_UNPACK), followed by up to 4 timestamps as indicated by the flags, followed by the key size and the key bytes.  If the WT_DELTA_IS_DELETE flag is not set, the entry then includes the value size and the value bytes.